### PR TITLE
debezium/dbz#1629 Schema evolution detection for DDL changes

### LINKED ISSUE #1629: CockroachDB connector: Schema evolution detection (DDL changes without restart)
## Summary

Automatically detect and adapt to DDL changes (ALTER TABLE ADD/DROP/RENAME COLUMN, etc.) without requiring a connector restart.

## Current Behavior

- `CockroachDBSchema.initialize()` queries `information_schema` **once at startup** and never refreshes
- If `ALTER TABLE ADD COLUMN` is executed while the connector runs, the changefeed JSON payload includes the new column but the connector **silently drops it** because the column is absent from the registered schema
- If `ALTER TABLE DROP COLUMN` is executed, the connector may fail or produce incorrect values
- README explicitly states: "No schema change detection: DDL changes are not automatically detected. Restart the connector after schema changes."

## CockroachDB Native Capability

CockroachDB changefeeds **handle schema changes natively**:
- When `ALTER TABLE ADD COLUMN` is executed, the changefeed performs a **backfill** -- it re-emits all existing rows with the new schema
- With `enriched_properties = 'schema'`, the changefeed includes the table schema in each event, providing a built-in mechanism to detect schema changes
- Events after the schema change automatically include new columns in the JSON payload

## Proposed Implementation

1. **Detect schema changes**: When processing changefeed events, compare incoming JSON field count/names against the registered `Table` column list. If fields don't match, trigger a schema refresh
2. **Schema refresh**: Re-query `information_schema.columns` and `information_schema.key_column_usage` for the affected table
3. **Update Debezium registry**: Call `tables().overwriteTable(newTable)` and `refreshSchema(tableId)` to rebuild the `TableSchema`
4. **Alternative approach**: Use `enriched_properties = 'schema'` in the changefeed to get schema metadata embedded in each event, avoiding round-trips to CockroachDB for schema discovery

## Acceptance Criteria

- [ ] `ALTER TABLE ADD COLUMN` detected and new column values emitted without restart
- [ ] `ALTER TABLE DROP COLUMN` handled gracefully (null values for removed columns during transition)
- [ ] `ALTER TABLE RENAME COLUMN` detected and reflected in emitted events
- [ ] Schema change emits appropriate schema change events to the schema change topic
- [ ] No data loss during schema transition
- [ ] Unit and integration tests for DDL scenarios

### PR DESCRIPTION
**Pull request by viragtripathi**
## Summary

Automatic schema evolution detection for DDL changes (ALTER TABLE ADD/DROP/RENAME COLUMN) without connector restart.

## Changes

- **CockroachDBSchema**: Added `refreshTable(connection, tableId)` to re-query `information_schema` and update registered table schema
- **CockroachDBStreamingChangeEventSource**: Added `hasSchemaChanged()` to compare incoming event fields against registered columns, and `refreshTableSchema()` to trigger schema refresh on mismatch
- **CockroachDBSchemaEvolutionTest**: 10 unit tests covering add/drop/rename column detection, edge cases
- **CockroachDBSchemaEvolutionIT**: Integration test verifying ALTER TABLE ADD COLUMN is detected and new field appears in emitted records without restart
- **README**: Added Schema Evolution Detection section

## How it works

1. Each incoming changefeed event's field names are compared against the registered table columns
2. On mismatch, `information_schema` is re-queried via a persistent connection
3. Internal schema is refreshed with `overwriteTable()` + `refreshSchema()`
4. Processing continues with the updated schema

CockroachDB changefeeds handle schema changes natively via backfill (re-emitting all rows with the new schema), so no events are lost.

## Test results

- 128 unit tests pass (118 existing + 10 new)
- Schema evolution IT passes (Testcontainers)
- 6 Cloud IT tests pass

Resolves https://github.com/debezium/dbz/issues/1629

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -34,7 +34,7 @@
         <!-- Debezium parent -->
         <version.debezium>${project.version}</version.debezium>
 
-        <cockroachdb.version>v26.1.0</cockroachdb.version>
+        <cockroachdb.version>v25.4.6</cockroachdb.version>
     </properties>
```
[the remark this answers is not in the document]
Comment (viragtripathi) [reply pom.xml:38]: this is intentional as we want to leave the LTS version as default i.e. v25.4.6
